app/channels/page_channel.rb

```diff
@@ -0,0 +1,32 @@
+class PageChannel < ApplicationCable::Channel
+  # TODO: store pages broadcasted by owner client in db
+  def subscribed
+    stream_from "page_channel_#{params[:notebook_id]}"
+  end
+
+  def receive(_data)
+    unless (page = persist_page)
+      # TODO: Handle scenario where save fails
+      puts "\n\n\nFailed to save page\n\n\n"
+    end
+    ActionCable.server.broadcast("page_channel_#{params[:notebook_id]}", page.layers)
+  end
+
+  def unsubscribed
+    # Any cleanup needed when channel is unsubscribed
+  end
+
+  private
+
+  def persist_page
+    @notebook = Notebook.find(params[:notebook_id].to_i)
+    page_number = Page.page.where(notebook_id: params[:notebook_id]).max_by(&:number).number + 1
+    page = Page.new(number: page_number, notebook: @notebook)
```

Comment:
Do we need to save the page here after we create it so it persists?

Reply:
Yes, this is being done for persistence. This allows the Notebook to reopen any added pages after the user navigates away.